### PR TITLE
bore: init at 0.3.3

### PR DESCRIPTION
**Pull request by delan**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] ~~Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))~~
- [ ] ~~Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`~~
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] ~~(Package updates) Added a release notes entry if the change is major or breaking~~
  - [ ] ~~(Module updates) Added a release notes entry if the change is significant~~
  - [ ] ~~(Module addition) Added a release notes entry if adding a new NixOS module~~
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by delan** [comment]
Not sure what to do about ofborg-eval-check-meta. I don’t understand the output, and I can’t find any docs about check-meta.nix or checkMeta. Any help would be much appreciated!

**Comment by delan** [comment]
It looks like my meta.maintainers was wrong, but I pretty much had to guess that this was the problem. Is there anything I read to help me understand future checkMeta failures?

File: pkgs/tools/networking/bore/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    | $out/bin/bore --decode \
+    | grep -q ';; NoError #0 Query 0 0 0 0 flags'
+  '';
+
+  meta = with lib; {
+    description = "DNS query tool";
+    homepage = "https://crates.io/crates/bore";
+    license = licenses.isc;
+    maintainers = [ maintainers.delan ];
+    mainProgram = "bore";
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:46]:
```suggestion
```
that is the default.

File: pkgs/tools/networking/bore/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  ];
+
+  buildInputs = lib.optional stdenv.isDarwin [
+    darwin.Libsystem
+    darwin.apple_sdk.frameworks.SystemConfiguration
+  ];
+
+  LIBCLANG_PATH="${llvmPackages.libclang.lib}/lib";
+
+  preFixup = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:30]:
```suggestion
  postInstall = ''
```

File: pkgs/tools/networking/bore/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  # FIXME can’t test --all-targets and --doc in a single invocation
+  cargoTestFlags = "--features std --all-targets --workspace";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional stdenv.isDarwin [
+    llvmPackages.libclang
+  ];
+
+  buildInputs = lib.optional stdenv.isDarwin [
+    darwin.Libsystem
+    darwin.apple_sdk.frameworks.SystemConfiguration
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:25]:
```suggestion
    Libsystem
    SystemConfiguration
```
please inherit those in callPackage.

File: pkgs/tools/networking/bore/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
+{ lib, stdenv, rustPlatform, fetchFromBitbucket, llvmPackages, darwin, installShellFiles }:
+rustPlatform.buildRustPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:2]:
```suggestion
{ lib, stdenv, rustPlatform, fetchFromBitbucket, llvmPackages, darwin, installShellFiles }:

rustPlatform.buildRustPackage rec {
```

**Comment by delan** [comment]
Thanks for the feedback @SuperSandro2000!

File: pkgs/tools/networking/bore/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  cargoBuildFlags = "-p ${pname}";
+
+  # FIXME can’t test --all-targets and --doc in a single invocation
+  cargoTestFlags = "--features std --all-targets --workspace";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional stdenv.isDarwin [
+    llvmPackages.libclang
+  ];
+
+  buildInputs = lib.optional stdenv.isDarwin [
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:24]:
```suggestion
  buildInputs = lib.optionals stdenv.isDarwin [
```

File: pkgs/tools/networking/bore/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+  cargoSha256 = "1n09gcp1y885lz6g2f73zw3fd0fmv7nwlvaqba2yl0kylzk7naa6";
+  cargoBuildFlags = "-p ${pname}";
+
+  # FIXME can’t test --all-targets and --doc in a single invocation
+  cargoTestFlags = "--features std --all-targets --workspace";
+
+  nativeBuildInputs = [ installShellFiles ] ++ lib.optional stdenv.isDarwin [
+    llvmPackages.libclang
+  ];
```
Comment (SuperSandro2000) [inline pkgs/tools/networking/bore/default.nix:22]:
```suggestion
  nativeBuildInputs = [ installShellFiles ]
    ++ lib.optional stdenv.isDarwin llvmPackages.libclang;
```

**Pull request merged**